lib/charms/kafka/v0/client.py

```diff
@@ -0,0 +1,298 @@
+        """
+        item_content = f"Message #{message_content}"
+        future = self._producer_client.send(self.topic, str.encode(item_content))
+        future.get(timeout=60)
+        logger.info(f"Message published to topic={self.topic}, message content: {item_content}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Handler for running a Kafka client")
+    parser.add_argument(
+        "-t",
+        "--topic",
+        help="Kafka topic provided by Kafka Charm",
+        type=str,
+        default="hot-topic",
+    )
+    parser.add_argument(
+        "-u",
+        "--username",
+        help="Kafka username provided by Kafka Charm",
+        type=str,
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        help="Kafka password provided by Kafka Charm",
+        type=str,
+    )
+    parser.add_argument(
+        "-c",
+        "--consumer-group-prefix",
+        help="Kafka consumer-group-prefix provided by Kafka Charm",
+        type=str,
+    )
+    parser.add_argument(
+        "-s",
+        "--servers",
+        help="comma delimited list of Kafka bootstrap-server strings",
+        type=str,
+    )
+    parser.add_argument(
+        "-x",
+        "--security-protocol",
+        help="security protocol used for authentication",
+        type=str,
+        default="SASL_PLAINTEXT",
+    )
+    parser.add_argument(
+        "-n",
+        "--num-messages",
+        help="number of messages to send from a producer",
+        type=int,
+        default=15,
+    )
+    parser.add_argument(
+        "-r",
+        "--replication-factor",
+        help="replcation.factor for created topics",
+        type=int,
+    )
+    parser.add_argument("--producer", action="store_true", default=False)
+    parser.add_argument("--consumer", action="store_true", default=False)
+    parser.add_argument("--cafile-path", type=str)
+    parser.add_argument("--certfile-path", type=str)
+    parser.add_argument("--keyfile-path", type=str)
+
+    args = parser.parse_args()
+    servers = args.servers.split(",")
+    if not args.consumer_group_prefix:
+        args.consumer_group_prefix = f"{args.username}-" if args.username else None
+
+    client = KafkaClient(
+        servers=servers,
+        username=args.username,
+        password=args.password,
+        topic=args.topic,
+        consumer_group_prefix=args.consumer_group_prefix,
+        security_protocol=args.security_protocol,
+        cafile_path=args.cafile_path,
+        certfile_path=args.certfile_path,
+        keyfile_path=args.keyfile_path,
+        replication_factor=args.replication_factor,
+    )
+
+    if args.producer:
+        logger.info(f"Creating new topic - {args.topic}")
+
+        topic = NewTopic(
+            name=args.topic,
+            num_partitions=args.num_partitions,
+            replication_factor=args.replication_factor,
+        )
+        client.create_topic(topic=topic)
```

Comment:
I find that topic handling could be a bit improved. I provide a suggestion in the general comment

Reply:
Yeah I think this is a good suggestion, lemme add it.